.github/workflows/reviewdog.yml

```
@@ -0,0 +1,73 @@
+name: Run Reviewdog
+on:
+  pull_request:
+permissions:
+  contents: read
+jobs:
+  analyze:
+    permissions:
+      contents: read
+      pull-requests: write
+    runs-on: ubuntu-24.04
+    steps:
+      - name: Install Harden-Runner
+        uses: step-security/harden-runner@0634a2670c59f64b4a01f0f96f84700a4088b9f0 # v2.12.0
+        with:
+          disable-sudo-and-containers: true
+          egress-policy: block
+          allowed-endpoints: >
+            api.adoptium.net:443
+            api.github.com:443
+            github-registry-files.githubusercontent.com:443
+            github.com:443
+            maven.pkg.github.com:443
+            objects.githubusercontent.com:443
+            raw.githubusercontent.com:443
+            repo.maven.apache.org:443
+      - name: Check out code and set up JDK and Maven
+        uses: s4u/setup-maven-action@4f7fb9d9675e899ca81c6161dadbba0189a4ebb1 # v1.18.0
+        with:
+          java-version: 17.0.13
+          java-distribution: temurin
+          maven-version: 3.9.9
+      - name: Set up Reviewdog
+        uses: reviewdog/action-setup@e04ffabe3898a0af8d0fb1af00c188831c4b5893 # v1.3.2
+        with:
+          # XXX: Configure Renovate to maintain this version.
+          reviewdog_version: v0.20.3
+      - name: Install project to local Maven repository
+        # XXX: Consider configuring Reviewdog such that it reports build
+        # errors during this step. When doing so, make sure that a nonzero
+        # `mvn` exit code does abort this workflow.
+        run: mvn -T1C install -DskipTests -Dverification.skip
+      - name: Build project with self-check and report issues
+        env:
+          REVIEWDOG_GITHUB_API_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        # XXX: Replace the Error Prone/javac diagnostics extraction with a more
+        # robust approach once available. See:
+        # - https://github.com/google/error-prone/issues/3766
+        # - https://github.com/google/error-prone/issues/444
+        # XXX: Cover additional output formats, besides Checkstyle and javac.
+        # XXX: Run with `-Dmaven.test.failure.ignore` instead of `-DskipTests`,
+        # and also report Surefire test failures.
+        run: |
+          mvn -T1C clean verify -Pself-check -DskipTests -Dverification.warn \
+            | reviewdog \
+                -name=javac \
+                -efm='[WARNING] %f:[%l,%c] %m' \
+                -filter-mode=file \
+                -reporter=github-pr-review \
+                -tee
+          find -name checkstyle-result.json -exec sh -c '
+            reviewdog -name=Checkstyle -f=sarif -reporter=github-pr-review < {}
+          ' \;
```

Ah, should perhaps be:
```suggestion
          find -name checkstyle-result.json -exec sh -c '
            reviewdog -name=Checkstyle -f=sarif -reporter=github-pr-review < "{}"
          ' \;
```